### ISSUE TITLE
Studypad does not open

### ISSUE THREAD

**Issue by gawako1953** [crosswire/xiphos#955]
I am using Debian testing with xiphos version 4.1.0.  When attempting to use studypad it fails silently.  All software updates have been applied with the last one done an hour or so before posting this.

I have tried multiple times to submit this to the Debian bts via reportbug but I cannot make it work with my ISP's email system.  I have also sent emails via my email client with the reportbug information copied and pasted into to the email address shown on the xiphos Package page on the Debian website,  it but those have been ignored.

**Comment by domcorbex** [comment]
On Debian Buster (and I guess last Ubuntu releases), Xiphos is built without the editor, you can't open the StudyPad nor edit the Personal commentary, see:
#794
and:
https://salsa.debian.org/pkg-crosswire-team/xiphos/commit/bb673775da79a47b3607077c693daa8dc22ddec7

**Comment by gawako1953** [comment]
On Sat, 2019-08-03 at 05:43 -0700, Dominique Corbex wrote:
> On Debian Buster (and I guess last Ubuntu releases), Xiphos is built
> without the editor, you can't open the StudyPad nor edit the Personal
> commentary, see:
> #794
> and:
> https://salsa.debian.org/pkg-crosswire-team/xiphos/commit/bb673775da79a47b3607077c693daa8dc22ddec7

Dominique,

Thank you for your replies, on each of the bugs I filed.  It's really
nice to have someone actually care enough to let me know what's going
on.

I find it incredible that the Debian team would build this version of
xiphos without Studypad and the editor. Those two features are why I
have used xiphos for years. This action on their part really limits my
use of xiphos as I use both features a lot and have a large number of
studies and commentary on scripture now inaccessible to me.    

I have tried to build xiphos from source using both Debian build tools
and the .waf method and have run into so many issues I finally gave up
on that.  I started compiling my kernels 15 years ago from source so
it's odd that I would run into so many problems that I just can't
figure out in building xiphos from source.  I think those issues are
due, at least in part, to the way Debian breaks up all their libraries
into individual packages so that their naming conventions are different
as I constantly run into the error of missing dev software and can find
no Debian packages by the name of the missing dev package errors
reported.  It's like with their python packages.  They are almost
without number.  

So, how would you suggest I go about getting a version of xiphos that I
can run on Buster that has the features I need?

**Comment by domcorbex** [comment]
gawako1953,
In fact, Xiphos editor is built upon WebKitGtk or GtkHTML libraries. Debian has dropped support for both, so eventually Deb devs built Xiphos without the editor.
libgtkhtml-editor-4.0-0 is available in Sid, maybe you can make a package for Buster and then build Xiphos as before.

**Comment by gawako1953** [comment]
On Sat, 2019-08-03 at 14:00 -0700, Dominique Corbex wrote:
> gawako1953,
> In fact, Xiphos editor is built upon WebKitGtk or GtkHTML libraries.
> Debian has dropped support for both, so eventually Deb devs built
> Xiphos without the editor.
> libgtkhtml-editor-4.0-0 is available in Sid, maybe you can make a
> package for Buster and then build Xiphos as before.
> 
> 

OK. That explains very well why the Debian xiphos package was built the
way it was. 

Do you happen to know if the Debian ziphos devs are going to build a
xiphos package in sid with the libgtkhtml-editor-4?  If so I would just
switch to sid as I ran that for years.  

Thanks once again for your help. I appreciate it as it gives me some
idea as to what I need to do as I don't care for the rpm system or foryum.

**Comment by domcorbex** [comment]
> Do you happen to know if the Debian ziphos devs are going to build a
> xiphos package in sid with the libgtkhtml-editor-4?
I don't think so.
Debian runs its own instance of GitLab, everyone can register an account (with an implicitly added suffix -guest):
https://salsa.debian.org/pkg-crosswire-team/xiphos
You can Salsa to check on progress being made on Deb packaging.

**Comment by gawako1953** [comment]
On Sat, 2019-08-03 at 14:00 -0700, Dominique Corbex wrote:
> gawako1953,
> In fact, Xiphos editor is built upon WebKitGtk or GtkHTML libraries.
> Debian has dropped support for both, so eventually Deb devs built
> Xiphos without the editor.
> libgtkhtml-editor-4.0-0 is available in Sid, maybe you can make a
> package for Buster and then build Xiphos as before.

> 
> 
I looked at this.  The libgtkhtml-editor-4.0-0 package in sid is, at
this moment anyway, for only the arm64 architecture according to the
page for that package.  I run an amd64 machine.  Porting to another
architecture type is waaay over my head.  The most I've ever done as
far as "programming" is writing bash and simple python scripts.  I am
not a developer.  

Anyway, thank you for all your time and help.  I appreciate it.

**Comment by LAfricain** [comment]
> In fact, Xiphos editor is built upon WebKitGtk or GtkHTML libraries. Debian has dropped support for both, so eventually Deb devs built Xiphos without the editor.

I have a question about this. Will the Xiphos dev team update this with some other libraries  or not?

**Comment by LAfricain** [comment]
@gawako1953 A work around is to use Bibletime.

**Comment by gawako1953** [comment]
I have the same question as with Debian dropping support for the editor
it drastically reduces the usability of Xiphos for many users as Debian
is the base for a lot of different distros.  Will Xiphos just tell
Debian-based users too bad, so sad, or what?

**Comment by gawako1953** [comment]
I have tried Bibletime and do not really like it.  It just feels really
clunky and doesn't have the same usability Xiphos does, at least to
me.  I don't like how the personal commentary works in it, and it
doesn't have a Studypad equivalent.

**Comment by LAfricain** [comment]
>doesn't have a Studypad equivalent.

Okay. But you can retrieve your notes that's already cool.

**Comment by gawako1953** [comment]
> Okay. But you can retrieve your notes that's already cool.

I actually tried Bibletime again and rediscovered, as it has been a few years since I tried using it, the real reason I stopped using it.  Bibletime regularly goes into a race condition and gets very slow.  I'd forgotten about that.  

If you open multiple tabs and then try to tile them vertically or horizontally it doesn't take long before Bibletime is using 100% of the cpu.  The qtwebkit(?) also goes nuts too, but I haven't seen it reach a full race condition.  Even just using it in the default configuration for a while results in a similar condition in which mousing over the text to bring up Strong's definitions gets extremely slow.  It gets so that there are lag times of 10 seconds and more for a new Strong's definition to appear in the Mag viewer..  It just gets unusable in a hurry.
